### PR TITLE
fix(api-client): harden response body preview against XSS

### PR DESCRIPTION
**Pull request by amritk**
## Problem

The response body preview component was vulnerable to XSS attacks and referrer leakage:

1. **XSS vectors**: Malicious `src` URLs using `javascript:`, `vbscript:`, or `data:text/html` protocols could execute arbitrary script in the app's origin
2. **Referrer leakage**: Media elements sent the user's location to untrusted third-party hosts via the `Referer` header
3. **Unsafe fallback**: The `<object>` element executes scripts inside embedded documents in the app's origin

## Solution

Implemented a defense-in-depth approach:

1. **URL validation**: Added `safeSrc` computed property that validates `src` against an allow-list of safe protocols:
   - `blob:` (standard `URL.createObjectURL` output)
   - `http:` / `https:`
   - `data:` URIs limited to known safe media types (`image/*`, `video/*`, `audio/*`, `application/pdf`, `application/octet-stream`)
   - Blocks `javascript:`, `vbscript:`, `file:`, malformed, and relative URLs

2. **Replaced `<object>` with sandboxed `<iframe>`**: The `<object>` fallback now renders as a fully sandboxed `<iframe>` with `sandbox=""` (disables scripts, forms, popups, and same-origin access), leaving only inert rendering like the browser's built-in PDF viewer

3. **Added referrer policy**: Set `referrerpolicy="no-referrer"` on all media elements (``, `<iframe>`) to prevent leaking the user's location to untrusted hosts

4. **Comprehensive test coverage**: Added 10 new security tests covering XSS vectors, allowed protocols, and sandbox enforcement. Updated existing tests to use realistic blob URLs and verify the new iframe behavior.

## Checklist

- [x] I explained why the change is needed.
- [x] I added a changeset.
- [x] I added tests (10 new security tests + updated 15 existing tests).
- [ ] I updated the documentation. (N/A — internal security hardening)

https://claude.ai/code/session_01QFfS3mHAZRQZZ3D7GWoQQF

---

> [!NOTE]
> **Medium Risk**
> Changes how untrusted response URLs are embedded (URL allow-listing + `<iframe>` sandboxing), which is security-sensitive and could alter preview rendering for some content types. Test coverage is expanded, but edge-case URL/media handling regressions are still possible.
> 
> **Overview**
> **Response body preview rendering is now hardened against XSS and referrer leakage.** `ResponseBodyPreview` introduces a `safeSrc` allow-list for `blob:`, `http(s):`, and constrained `data:` media URIs, and shows *“Preview unavailable”* when the `src` is unsafe or malformed.
> 
> The generic document preview switches from `<object>` to an `<iframe>`, applying `sandbox=""` for non-PDF content (PDFs are exempt) and adds `referrerpolicy="no-referrer"` to `img`, `video`, `audio`, and the iframe. Tests are updated accordingly and expanded with dedicated security cases, and a patch changeset is added.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit f1ab2263bc98dbc02f09c103269392dcdc317e01. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

**Review by DemonHa — APPROVED** [review]
GREAT 😊

File: packages/api-client/src/v2/blocks/response-block/components/ResponseBodyPreview.vue [hunk 1]
```diff
@@ -46,40 +94,57 @@ watch(
...
+    <iframe
       v-else
-      class="aspect-[4/3] w-full"
-      :data="src"
-      :type="type"
-      @error="error = true" />
+      class="aspect-[4/3] w-full border-0"
+      :src="safeSrc"
+      referrerpolicy="no-referrer"
+      sandbox="" />
```
[the remark this answers is not in the document]
Comment (amritk) [reply packages/api-client/src/v2/blocks/response-block/components/ResponseBodyPreview.vue]: a pdf can also execute js escaping the iframe, so lets leave this and config it in if someone wants it

[the remark this answers is not in the document]
Comment (amritk) [reply packages/api-client/src/v2/blocks/response-block/components/ResponseBodyPreview.vue]: oh nvm I stand corrected
